cruce.cpp

```diff
@@ -0,0 +1,100 @@
+#include<fstream>
+using namespace std;
+ifstream fin("cruce.in");
+ofstream fout("cruce.out");
+long long columnSum[501]={0};
+long long lineSum[501]={0};
+void read(int &n, int &k, int matrix[501][501])
+{
+    fin>>n>>k;
+    for(int index=1; index<=n; ++index)
```

Comment:
Desi nu este incorect, de obicei preferam sa avem indicii de la 0. Astfel ciclul `for` poate fi simplificat:
```
for(int index = 0; index<n; ++index)
```
Intregul program ar putea fi scris:
```
for(int l = 0; l<n; ++l) {
  for(int c = 0; c<n; ++c) {
    fin >> matrix[l][c];
    lineSum[l] += matrix[l][c];
    columnSum[c] += matrix[l][c];
  }
}  
```
Diferenta nu este foarte mare, dare este in pic mai usor de citit.

Reply:
Da, acolo la indici, așa se cere in problema. In mod normal folosesc și eu indexarea de la 0.